body.tex

```diff
@@ -1,166 +1,466 @@
+Potential optimizations and further investigations were identified throughout the PoC project but were not carried out.
+Some ideas are described in Sec \ref{future}.
 
-\begin{itemize}
-\item LSST Catalogs (CatSim): \cite{2014SPIE.9150E..14C}
-\item Feature-Based Scheduler: \cite{2018arXiv181004815N}
-\item Operations Simulator (OpSim): Scheduler \cite{2016SPIE.9910E..13D}, SOCS \cite{2016SPIE.9911E..25R}
-\item Metrics Analysis Framework (MAF): \cite{2014SPIE.9149E..0BJ}
-\item Image simulations (Phosim): \cite{2015ApJS..218...14P}
-\item Sky brightness model: \cite{2016SPIE.9910E..1AY}
-\item LSST Performance for NEO (or moving object) discovery: \cite{2018Icar..303..181J}
-\end{itemize}
-%------------------------------------------------------------------------------
 
+\section{Building AWS support into the Data Butler}
+\label{sec:butler}
 
-\subsubsection{Data Management}
+The Data Butler is the overarching data IO abstraction through which all LSST data access is mediated \citeds{DMTN-056}. Datasets are referred to by their unique IDs, or a set of identifying references, which are then resolved through a Registry that matches the dataset IDs, or references, to the location, file format and the Python object type of the dataset. The system that persists, reads and potentially modifies the datasets is called the Datastore. The Registry is almost always backed by a SQL database and the Datastore is usually backed by a shared filesystem. A major focus of the \poc~was to implement, and investigate issues related to, an S3 backed Datastore and a PostgreSQL backed Registry.
 
-LSST data management system and the data products are described in:
+At the time the \poc~project began, Generation 3 Data Butler implemented \texttt{PosixDatastore}, a local or shared filesystem datastore, and a SqliteRegistry. OracleRegistry followed soon after the \poc~work began. Initially the focus was on implementing an S3 backed datastore called S3Datastore. The interface between AWS services and LSST Stack would be based on the official AWS SDK called boto3. In March 2019, Dino Bektesevic visited Tim Jenness for this work and implemented the early versions of a new module in the \texttt{daf\_butler} called \texttt{s3utils}, an \texttt{S3Datastore} class, the \texttt{PosixDatastore} equivalent, and a set of appropriate unit tests that demonstrated its functionality and correctness. The unit tests for the datastore utilize the \texttt{moto} library which mocks requests and responses sent to AWS services, so that no additional external infrastructure is required to use it. \texttt{PostgreSqlRegistry} class was implemented partially during the visit and completed shortly after the visit. The initial implementation showcasing the required changes to the code was submitted as a Draft Pull Request \href{https://github.com/lsst/daf_butler/pull/147}{PR-147} in \texttt{daf\_butler}.
```

Comment:
In this section, I think it's a bit excessive to essentially document ticket-by-ticket changes to the code base, along with specific meetings and PRs.  These details could be moved to an appendix or to footnotes (e.g. for the PRs); this section should summarize.

Reply:
I see @hsinfang already pruned this section - thanks for that(!) and sorry for the late response to the PR, but like I said bellow I must've missed the notifications. 

I was asked to document everything I did - so I did. I can prune further if required?

Reply:
(So far I haven't reorganized the session, so I'm leaving this unresolved for now)

Reply:
I moved the Butler section to the Appendix. For now I don't see an individual short Butler section is still needed; instead I expanded the Tech Stack section and the Architecture section where Butler ideas are introduced and explained. 